crosshair/register_contract.py

```diff
@@ -0,0 +1,104 @@
+"""API for registering contracts for external libraries."""
+from dataclasses import dataclass
+from inspect import Parameter, Signature, ismethod, signature
+from typing import Callable, Dict, Optional
+
+from crosshair.stubs_parser import signature_from_stubs
+from crosshair.util import debug
+
+
+class ContractRegistrationError(Exception):
+    pass
+
+
+@dataclass
+class Contract:
```

Comment:
Can we name this something like `ContractOverride`? Just to be extra clear that this isn't the class CrossHair directly uses to check contracts.

Reply:
I totally agree with that, I will rename it soon!